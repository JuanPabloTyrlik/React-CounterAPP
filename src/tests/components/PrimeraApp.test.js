import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import PrimeraApp from '../../components/PrimeraApp';
// import {render} from '@testing-library/react';

describe('Pruebas en <PrimeraApp />', () => {
    // test('should render "Hola, Soy Goku', () => {
    //     const saludo = 'Hola, Soy Goku';
    //     const wrapper = render(<PrimeraApp saludo={saludo} />);
    //     expect(wrapper.getByText(saludo)).toBeInTheDocument();
    // })
    test('should render <PrimeraApp /> correctly', () => {
        const saludo = 'Hola, Soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render the subtitle sent with props', () => {
        const saludo = 'Hola, Soy Goku';
        const subtitulo = 'Subtitulo';
        const wrapper = shallow(
            <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
        );
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
    });
});
