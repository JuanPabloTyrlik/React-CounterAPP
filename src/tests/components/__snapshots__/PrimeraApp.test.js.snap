// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> should render <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    Mi primera aplicaci√≥n
  </p>
</Fragment>
`;
